src/app/space/app-launcher/services/app-launcher-che.service.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { TestBed } from '@angular/core/testing';
+import { Observable, of } from 'rxjs';
+import { Che } from './../../create/codebases/services/che';
+import { AppLaunchCheService } from './app-launcher-che.service';
+
+describe('AppLaunchCheService: ', () => {
+  const mockCheState: Che = {
+    clusterFull: false,
+    multiTenant: true,
+    running: true
+  };
+
+  const mockAppLauncherCheService = {
+    getState(): Observable<Che> {
+      return of(mockCheState);
+    }
+  };
+
+  let appLaunchCheService: AppLaunchCheService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: AppLaunchCheService, useValue: mockAppLauncherCheService }
+      ]
+    });
+    appLaunchCheService = TestBed.get(AppLaunchCheService);
+  });
+
+  it('Get che state', async () => {
+    appLaunchCheService.getState().subscribe((val: Che) => {
+      expect(val).toEqual(mockCheState);
```

Comment:
Not sure what's the goal of this test. Are we testing that mock is returned? Can we think about some higher level functionality/behaviour instead of making sure we know how to pass values and mocks around? :)

Reply:
There are already tests for ```cheService``` in fabric8-ui. You can find it here https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/services/che.service.spec.ts